src/glang/codegen/translation_unit.py

```diff
@@ -180,22 +230,59 @@ def add_static_variable(self, var: StaticVariable) -> None:
     def add_function_body(self, function: Function) -> None:
         self._fn_bodies.append(function)
 
-    def generate_ir(self) -> list[str]:
-        lines: list[str] = []
+    def add_secondary_file(self, path: Path) -> DIFile:
+        di_file = DIFile(next(self._metadata_gen), path)
+        self.di_files.append(di_file)
+        return di_file
+
+    def generate_ir(self) -> str:
+        output = IROutput(metadata={self.di_compile_unit, *self.di_files})
 
-        lines.append(f'target datalayout = "{target.get_datalayout()}"')
-        lines.append(f'target triple = "{target.get_target_triple()}"')
+        output.lines.append(f'target datalayout = "{target.get_datalayout()}"')
+        output.lines.append(f'target triple = "{target.get_target_triple()}"')
 
-        lines.append("")
-        var_reg_gen = count(0)
+        output.lines.append("")
+        # FIXME the file passed in is wrong (although we aren't using it yet).
+        ctx = IRContext(count(0), count(0), self.di_files[0])
         for variable in self._static_variables:
-            lines.extend(variable.generate_ir(var_reg_gen))
+            output.extend(variable.generate_ir(ctx))
 
-        lines.append("")
-        lines.extend(self.symbol_table.get_ir_for_initialization())
+        output.lines.append("")
+        output.lines.extend(self.symbol_table.get_ir_for_initialization())
 
-        lines.append("")
+        # We need to declare any debugger intrinsics we use for some reason.
+        output.lines.append("")
+        output.lines.append(
+            "declare void @llvm.dbg.declare(metadata, metadata, metadata)"
+        )
+
+        output.lines.append("")
         for func in self._fn_bodies:
-            lines.extend(func.generate_ir())
+            output.extend(func.generate_ir(self._metadata_gen))
+
+        # behaviour = 1 (Error)
+        # Emits an error if two values disagree, otherwise the resulting value
+        # is that of the operands.
```

Comment:
Not sure I understand this comment tbh...
Oh `MetadataFlag.behaviour = 1`... when is the error emitted, is this when the debugger attaches?

Reply:
Sorry I was being lazy. I made it an enum now.

Basically, this behavior flag determines how LLVM deals with conflicting metadata when merging two or more "modules" together.

Error means that compilation/linking should fail.